src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplicationBase.cs

```diff
@@ -0,0 +1,208 @@
+﻿/*
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    /// <summary>
+    /// The class for supporting multi-components based application model.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class ComponentBasedApplicationBase : Application
+    {
+        private const string LogTag = "Tizen.Applications.ComponentBasedApplicationBase";
+        private IList<ComponentFactoryBase> _componentFactories = new List<ComponentFactoryBase>();
+        private Interop.CBApplication.CBAppLifecycleCallbacks _callbacks;
+        private bool _disposedValue = false;
+
+        /// <summary>
+        /// Initializes the ComponentBasedApplicationBase class.
+        /// </summary>
+        /// <param name="typeInfo">The component type information.
+        /// The key should be a class type of BaseComponent subclass.
+        /// The value should be a component id which is declared in tizen-manifest.xml.
+        /// </param>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of out of memory.</exception>
```

Comment:
Do not throw OutOfMemoryException on your code. 

https://docs.microsoft.com/ko-kr/dotnet/standard/design-guidelines/using-standard-exception-types#outofmemoryexception

Reply:
Done